app/controllers/admins/semesters_controller.rb

```diff
@@ -0,0 +1,62 @@
+module Admins
+  class SemestersController < BaseController
+    def index
+      @semesters = Semester.all
+    end
+
+    def show
+      @semester = Semester.find params[:id]
+    end
+
+    def new
+      @semester = Semester.new
+    end
+
+    def create
+      @semester = Semester.new member_params
+      clear_current_semester
+      if @semester.save
+        redirect_to admin_semesters_path
+      else
+        render 'new'
+      end
+    end
+
+    def edit
+      @semester = Semester.find params[:id]
+    end
+
+    def update
+      @semester = Semester.find params[:id]
+      clear_current_semester
+      if @semester.update_attributes member_params
+        redirect_to admin_semesters_path
+      else
+        render 'edit'
+      end
+    end
+
+    def clear_current_semester
+      return unless member_params[:is_current_semester]
+      Semester.where('is_current_semester').update_all("is_current_semester = 'false'")
+    end
```

Comment:
This code is probably better either in the Semester model itself or in a service object since it concerns only the semester model (whereas code in controllers is typically used to prepare variables for views).